app/src/main/java/org/worldbank/transport/driver/utilities/DriverUtilities.java

```diff
@@ -217,9 +221,36 @@ public static boolean localeIsRTL() {
     }
 
     public static boolean localeIsRTL(Locale locale) {
-        Log.d(LOG_LABEL, locale.getDisplayName());
+        Log.d(LOG_LABEL, "Locale " + locale.getDisplayName() + " language " + Locale.getDefault().getLanguage());
+
         int direction = Character.getDirectionality(locale.getDisplayName().charAt(0));
         return  direction == Character.DIRECTIONALITY_RIGHT_TO_LEFT_ARABIC ||
                 direction == Character.DIRECTIONALITY_RIGHT_TO_LEFT;
     }
+
+    /**
+     * Check if the system language is Arabic. Use to decide whether to use Hijri date display.
+     *
+     * @return True if device language is Arabic.
+     */
+    public static boolean languageIsArabic() {
+        return Locale.getDefault().getLanguage().startsWith("ar");
+    }
+
+    /**
+     * Convert a date to its Hijri calendar representation.
+     * Uses Joda for the conversion, as it is already included as a dependency.
+     *
+     * @param date ISO date/time to format
+     * @return Hijri date string
+     */
+    public static String formatDateAsHijri(Date date) {
+        final Chronology hijri = IslamicChronology.getInstanceUTC();
```

Comment:
Ok, good to know.

Looks like the latest API (24) has added support for [Islamic calendar types to Android](https://developer.android.com/reference/android/icu/util/IslamicCalendar.html), but that's not going to help us here.

I might give [this project](https://github.com/msarhan/ummalqura-calendar) a try.